src/core/meshcop/commissioner.cpp

```diff
@@ -120,17 +171,17 @@ otError Commissioner::Stop(void)
 
     Get<Coap::CoapSecure>().Stop();
 
-    mState = OT_COMMISSIONER_STATE_DISABLED;
     Get<ThreadNetif>().RemoveUnicastAddress(mCommissionerAloc);
     Get<Notifier>().Signal(OT_CHANGED_COMMISSIONER_STATE);
     RemoveCoapResources();
     ClearJoiners();
     mTransmitAttempts = 0;
 
     mTimer.Stop();
-
     Get<Coap::CoapSecure>().Stop();
 
+    SetState(OT_COMMISSIONER_STATE_DISABLED);
+
     SendKeepAlive();
```

Comment:
Same here? Can we `SetState()` at the very end?

Reply:
Current implementation of `SendKeepAlive()` depends on `mState` to determine the contents of the keep alive message.